sdk/dotnet/Pulumi/Core/InputMap.cs

```diff
@@ -62,6 +62,33 @@ public void Add(string key, Input<V> value)
             set => Add(key, value);
         }
 
+        /// <summary>
+        /// Merge two instances of <see cref="InputMap{V}"/>. Returns a new <see cref="InputMap{V}"/>
+        /// without modifying any of the arguments. 
+        /// <para/>If both maps contain the same key, the value from the second map takes over.
+        /// </summary>
+        /// <param name="first">The first <see cref="InputMap{V}"/>. Has lower priority in case of
+        /// key clash.</param>
+        /// <param name="second">The second <see cref="InputMap{V}"/>. Has higher priority in case of
+        /// key clash.</param>
+        /// <returns>A new instance of <see cref="InputMap{V}"/> that contains the items from
+        /// both input maps.</returns>
+        public static InputMap<V> Merge(InputMap<V> first, InputMap<V> second)
+        {
+            var firstDictionary = (Input<ImmutableDictionary<string, V>>)first._outputValue;
+            var secondDictionary = (Input<ImmutableDictionary<string, V>>)second._outputValue;
```

Comment:
these casts seem unnecessary.  you should have two `Output<...>`s. Those should both work with Output.Tuple afaict.

Reply:
You are totally right, I confused myself. Fixed!